plugins/autosuspend.koplugin/main.lua

```diff
@@ -169,87 +172,97 @@ function AutoSuspend:onPreventStandby()
+            self:setSuspendShutdownTimes(touchmenu_instance,
+                _("Timeout for autosuspend"), _("Enter time in hours and minutes."),
+                "auto_suspend_timeout_seconds", default_auto_suspend_timeout_seconds,
+                {60, 24*3600}, false)
         end,
     }
     if not (Device:canPowerOff() or Device:isEmulator()) then return end
     menu_items.autoshutdown = {
         sorting_hint = "device",
         text_func = function()
             if self.autoshutdown_timeout_seconds  then
-                local duration_format = G_reader_settings:readSetting("duration_format", "classic")
-                return T(_("Autoshutdown timeout: %1"),
-                    util.secondsToClockDuration(duration_format, self.autoshutdown_timeout_seconds, true))
+                local time_string = util.secondsToClockDuration("modern",
+                    self.autoshutdown_timeout_seconds, true, true, true):gsub("00m$","")
+                return T(_("Autoshutdown timeout: %1"), time_string)
             else
                 return _("Autoshutdown timeout")
             end
         end,
         keep_menu_open = true,
         callback = function(touchmenu_instance)
-            local InfoMessage = require("ui/widget/infomessage")
-            local SpinWidget = require("ui/widget/spinwidget")
-            local autosuspend_spin = SpinWidget:new {
-                value = self.autoshutdown_timeout_seconds / 60 / 60,
-                -- About a minute, good for testing and battery life fanatics.
-                -- Just high enough to avoid an instant shutdown death scenario.
-                value_min = 0.017,
-                -- More than three weeks seems a bit excessive if you want to enable authoshutdown,
-                -- even if the battery can last up to three months.
-                value_max = 28*24,
-                value_hold_step = 24,
-                precision = "%.2f",
-                ok_text = _("Set timeout"),
-                title_text = _("Timeout in hours"),
-                callback = function(autosuspend_spin)
-                    self.autoshutdown_timeout_seconds = math.floor(autosuspend_spin.value * 60 * 60)
-                    G_reader_settings:saveSetting("autoshutdown_timeout_seconds", self.autoshutdown_timeout_seconds)
-                    UIManager:show(InfoMessage:new{
-                        text = T(_("The system will automatically shut down after %1 hours of inactivity."),
-                            string.format("%.2f", self.autoshutdown_timeout_seconds / 60 / 60)),
-                        timeout = 3,
-                    })
-                    self:_unschedule()
-                    self:_start()
-                    if touchmenu_instance then touchmenu_instance:updateItems() end
-                end
-            }
-            UIManager:show(autosuspend_spin)
+            -- 5*60 sec (5') is the minimum and 28*24*3600 (28days) is the maximum shutdown time.
```

Comment:
We lost the reasoning from the old comments here.

Reply:
I have added some comments in the next commit. Hope these are ok.